zend_abstract_interface/hook/hook.c

```diff
@@ -132,26 +127,37 @@ static int zai_hook_resolve_impl(zval *zv ZAI_TSRMLS_DC) {
 
     if (!function) {
         /* cannot be resolved */
-        return ZEND_HASH_APPLY_KEEP;
+        return false;
     }
 
     HashTable *table;
 
     if (!zai_hook_resolved_table(zai_hook_install_address(function), &table ZAI_TSRMLS_CC)) {
-        return ZEND_HASH_APPLY_KEEP;
+        return false;
     }
 
     zai_hook_t *resolved;
 
     if (!zai_hook_table_insert(table, hook, sizeof(zai_hook_t), (void **)&resolved)) {
-        return ZEND_HASH_APPLY_KEEP;
+        return false;
     }
 
-    zai_hook_copy(resolved ZAI_TSRMLS_CC);
-
     zai_hook_memory_reserve(resolved);
 
-    return ZEND_HASH_APPLY_REMOVE;
+    return true;
+}
+
+#if PHP_VERSION_ID < 70000
+static int zai_hook_resolve_impl(zai_hook_t *hook ZAI_TSRMLS_DC) {
+#else
+static int zai_hook_resolve_impl(zval *zv ZAI_TSRMLS_DC) {
+    zai_hook_t *hook = Z_PTR_P(zv);
+#endif
+    if (zai_hook_resolve_hook(hook ZAI_TSRMLS_CC)) {
+        zai_hook_copy(hook ZAI_TSRMLS_CC);
```

Comment:
Copy and remove looks wrong.

Reply:
It's what has been happening before though, look at line 150 in the previous version:

https://github.com/DataDog/dd-trace-php/blob/cdc6833f3f34d52a35ed46b64282de1bd6c0c3ac/zend_abstract_interface/hook/hook.c#L150

Reply:
I actually misread that, I read literally "remove" and "copy", not "resolve" and "copy" ... my mistake.